File: Programowanie Obiektowe Projekt Nr 3 Python/Turtle.py
#-------------------------------------------------------------------------------
# Name:        module1
# Purpose:
#
# Author:      Kontekstowy
#
# Created:     27-05-2016
# Copyright:   (c) Kontekstowy 2016
# Licence:     <your licence>
#-------------------------------------------------------------------------------
from Animal import Animal
class Turtle(Animal):
    def __init__(self,xx,yy,world,numberx,tempSign):
        Animal.__init__(self,xx,yy,world,numberx,tempSign)
        self._initiative=1
        self._strength=2
        self._sign='T'

    def action(self,animalBoard,nx,ny,animalContainer,board):
        if animalBoard[self._x  + nx][self._y +ny] == -1:
            return 1
        if animalBoard[self._x + nx][self._y + ny] != -1:
            if animalBoard[self._x + nx][self._y + ny] == 'W':
                print("Zolw ginie atakujac wilka")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == 'F':
                print("Zolw ginie atakujac lisa")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == 'T':
                if nx !=0 and ny !=0:
                    return 4
                else:
                    return 0
            if animalBoard[self._x + nx][self._y + ny] == 'A':
                print("Zolw ginie atakujac antylope")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == 'H':
                print("Zolw ginie atakujac czlowieka")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == '#':
                print("Zolw zjada trawe")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'G':
                print("Zolw zjada Guarane")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'N':
                print("Zolw zjada trujace wilcze jagody i ginie")
                return 5
            if animalBoard[self._x + nx][self._y + ny] == 'L':
                print("Zolw zjada mleczyk")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'S':
                print("Zolw zjada owce")
                return 2

File: Programowanie Obiektowe Projekt Nr 2 Java/src/objectiveworld160553/Turtle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objectiveworld160553;

/**
 *
 * @author Kontekstowy
 */
public class Turtle extends Animal {

    public Turtle(int x, int y, World world) {
        super(x, y, world);
        this.name = "T";
        this.initiative = 1;
        this.strength = 2;
    }

         @Override
    public int action(char animalBoard[][], int nx, int ny, Organism me) {
        switch (animalBoard[x + nx][y + ny]) {
            case '0':
                return 1;
            case 'T':
        //        System.out.println("Spotkaly sie 2 zolwie");
                return 4;
                case 'H':
                System.out.println("Zolw ginie atakujac czlowieka");
                return 3;
            case 'S':
                System.out.println("Owca zabija atakujacego zolwia");
                return 3;
            case 'N':
                System.out.println("Zolw zjada wilcze jagody i ginie");
                return 5;
            case 'W':
                System.out.println("Wilk zabija atakujacego zolwia");
                return 3;
            case 'F':
                System.out.println("Lis zabija atakujacego zolwia");
                return 3;
            case 'L':
                System.out.println("Lis zjada mlecz");
                return 2;
            case '#':
                System.out.println("Lis zjada trawe");
                return 2;
            case 'G':
                System.out.println("Lis zjadla guaraney");
                return 2;
        }

        return 0;
    }

}
